internal/command/views/test_test.go

```diff
@@ -3078,6 +3255,320 @@ func TestTestJSON_FatalInterruptSummary(t *testing.T) {
+					&states.ResourceInstanceObjectSrc{
+						Status: states.ObjectReady,
+					},
+					addrs.AbsProviderConfig{
+						Module:   addrs.RootModule,
+						Provider: addrs.NewDefaultProvider("test"),
+					})
+				state.SetResourceInstanceDeposed(
+					addrs.Resource{
+						Mode: addrs.ManagedResourceMode,
+						Type: "test",
+						Name: "bar",
+					}.Instance(addrs.NoKey).Absolute(addrs.RootModuleInstance),
+					"0fcb640a",
+					&states.ResourceInstanceObjectSrc{
+						Status: states.ObjectReady,
+					},
+					addrs.AbsProviderConfig{
+						Module:   addrs.RootModule,
+						Provider: addrs.NewDefaultProvider("test"),
+					})
+			}),
+			stderr: "",
+			want: []map[string]interface{}{
+				{
+					"@level":   "info",
+					"@message": "Writing state to file: errored_test.tfstate",
+					"@module":  "tofu.ui",
+				},
+			},
+		},
+		"state_null_resource_with_errors": {
+			file: &moduletest.File{Name: "main.tftest.hcl"},
+			state: states.BuildState(func(state *states.SyncState) {
+				state.SetResourceInstanceCurrent(
+					addrs.Resource{
+						Mode: addrs.ManagedResourceMode,
+						Type: "null_resource",
+						Name: "failing_will_depend_on_me",
+					}.Instance(addrs.NoKey).Absolute(addrs.RootModuleInstance),
+					&states.ResourceInstanceObjectSrc{
+						Status: states.ObjectReady,
+					},
+					addrs.AbsProviderConfig{
+						Module:   addrs.RootModule,
+						Provider: addrs.NewDefaultProvider("null"),
+					})
+				state.SetResourceInstanceCurrent(
+					addrs.Resource{
+						Mode: addrs.ManagedResourceMode,
+						Type: "null_resource",
+						Name: "failing",
+					}.Instance(addrs.NoKey).Absolute(addrs.RootModuleInstance),
+					&states.ResourceInstanceObjectSrc{
+						Status: states.ObjectReady,
+						Dependencies: []addrs.ConfigResource{
+							{
+								Module: []string{},
+								Resource: addrs.Resource{
+									Mode: addrs.ManagedResourceMode,
+									Type: "null_resource",
+									Name: "failing_will_depend_on_me",
+								},
+							},
+						},
+						CreateBeforeDestroy: false,
+					},
+					addrs.AbsProviderConfig{
+						Module:   addrs.RootModule,
+						Provider: addrs.NewDefaultProvider("null"),
+					})
+			}),
+			stderr: "",
+			want: []map[string]interface{}{
+				{
+					"@level":   "info",
+					"@message": "Writing state to file: errored_test.tfstate",
+					"@module":  "tofu.ui",
+				},
+			},
+		},
+	}
+	// Run tests for Human view
+	runTestSaveErroredStateFile(t, tcsHuman, arguments.ViewHuman)
+
+	// Run tests for JSON view
+	runTestSaveErroredStateFile(t, tcsJson, arguments.ViewJSON)
+}
+
+func runTestSaveErroredStateFile(t *testing.T, tc map[string]struct {
+	state  *states.State
+	run    *moduletest.Run
+	file   *moduletest.File
+	stderr string
+	want   interface{}
+}, viewType arguments.ViewType) {
+	for name, data := range tc {
+		t.Run(name, func(t *testing.T) {
+			// Create a temporary directory
+			tempDir := t.TempDir()
```

Comment:
You're creating this temporary directory, but I don't see any place where this is cleaned up.

That makes sense for the actual running (obviously, we want people to be able to inspect the statefile, that's the point), but for unit tests we definitely need to clean up, or each unit test run will create more files on the users system.

Reply:
This is actually good, it's a testing-specific TempDir and is cleaned up automatically.